core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -744,7 +750,7 @@ private[spark] class BlockManager(
       level: StorageLevel,
       tellMaster: Boolean = true,
       effectiveStorageLevel: Option[StorageLevel] = None,
-      keepReadLock: Boolean = false): Option[PutResult] = {
+      keepReadLock: Boolean = false): Option[Option[Iterator[Any]]] = {
```

Comment:
I checked the callsites. Currently we never actually use the inner level of option. The only place where we look at the `Some` value is in `getOrElseUpdate`. There we just do `x.get.get` anyway, so this can just be an `Option[Iterator[Any]]`.

If there's another boolean that we need to return, then I would just make a private case class even if it's called in only 1 place. Right now it's unclear what `Some(None)` means, for example. Anything we can think of here is better than nested options.